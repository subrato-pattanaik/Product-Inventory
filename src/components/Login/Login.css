.login {
  min-height: 89vh;
}
