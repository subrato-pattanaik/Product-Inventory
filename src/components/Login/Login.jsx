import React, { useContext } from "react";
import { withRouter, useHistory, useLocation } from "react-router-dom";
import { Container, Row, Col, Form, Button, Alert } from "react-bootstrap";
import { useForm } from "react-hook-form";
import "./Login.css";
import login from "../../images/image1.jpg";
import * as yup from "yup";
import { yupResolver } from "@hookform/resolvers/yup";
import axios from "axios";
import { UserContextAPI } from "../ContextAPI/UserContextAPI";

const schema = yup.object().shape({
  email: yup.string().required("*Email is required for login"),
  password: yup.string().required("*Please enter password"),
});

function Login() {
  const { user, setUser } = useContext(UserContextAPI);

  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  const image = {
    backgroundImage: `url(${login})`,
    backgroundSize: "cover",
    backgroundPosition: "center center",
    minHeight: "89vh",
  };

  let navigation = useHistory(); //for navigation
  let location = useLocation();

  const checkExistence = (serverUsers, formData) => {
    const user = serverUsers.find((user) => user.email === formData.email);
    if (user) return user;
  };

  const onSubmit = async (formData, event) => {
    const users = await axios
      .get("http://localhost:4000/users")
      .then((response) => checkExistence(response.data, formData));

    if (!users)
      alert(
        "Account for this E-Mail doesn't exit !! \nCreate a new account \nOR \nYou might have typos in E-Mail"
      );
    else {
      if (users.password === formData.password) {
        event.target.reset();
        setUser(users.id);
        navigation.push("/productList");
      } else alert("Wrong password");
    }
  };

  return (
    <>
      {!!user ? (
        <>{navigation.push("/")}</>
      ) : (
        <Container fluid={true} className="p-0">
          <Row noGutters>
            <Col lg={6} md={6} xs={12} style={image}></Col>
            <Col lg={6} md={6} xs={12}>
              <Container className="d-flex login bg-light align-items-center">
                <Row className="mx-auto justify-content-center">
                  <Col md={10} xs={10}>
                    {!!location?.state?.pass && (
                      <Alert variant="success">
                        <Alert.Heading>
                          Successfully Registered !!!
                        </Alert.Heading>
                      </Alert>
                    )}
                    {!!location?.state?.addProduct && (
                      <Alert variant="danger">
                        <Alert.Heading>
                          Login First then add the product
                        </Alert.Heading>
                      </Alert>
                    )}
                    {!!location?.state?.editProduct && (
                      <Alert variant="danger">
                        <Alert.Heading>
                          Login First then edit the product
                        </Alert.Heading>
                      </Alert>
                    )}
                    {!!location?.state?.deleteproduct && (
                      <Alert variant="danger">
                        <Alert.Heading>
                          Login First then delete the product
                        </Alert.Heading>
                      </Alert>
                    )}

                    <h3 className="display-4 text-warning text-center font-weight-bold">
                      PI Login
                    </h3>
                    <p className="text-muted text-center mb-4">
                      Lets do shopping from here with best deals
                    </p>
                    <Form autoComplete="off" onSubmit={handleSubmit(onSubmit)}>
                      <Form.Group controlId="formBasicEmail" className="mb-4">
                        <Form.Label className="small lead">
                          Email address
                        </Form.Label>
                        <Form.Control
                          className="rounded-pill"
                          type="email"
                          placeholder="Enter email"
                          name="email"
                          ref={register}
                        />
                      </Form.Group>
                      {errors.email && (
                        <div className="small lead text-danger mt-n3 mb-4">
                          {errors.email.message}
                        </div>
                      )}
                      <Form.Group
                        controlId="formBasicPassword"
                        className="mb-4"
                      >
                        <Form.Label className="small lead">Password</Form.Label>
                        <Form.Control
                          className="rounded-pill"
                          type="password"
                          name="password"
                          placeholder="Password"
                          ref={register}
                        />
                        {errors.password && (
                          <div className="small lead text-danger mb-n1 mb-4">
                            {errors.password.message}
                          </div>
                        )}
                      </Form.Group>

                      <Button
                        variant="warning"
                        type="submit"
                        className="btn-block text-white border-0 rounded-pill mb-4"
                      >
                        Login here
                      </Button>
                    </Form>
                  </Col>
                </Row>
              </Container>
            </Col>
          </Row>
        </Container>
      )}
    </>
  );
}

export default withRouter(Login);
