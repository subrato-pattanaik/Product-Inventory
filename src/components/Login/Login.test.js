import React from "react";
import { mount } from "enzyme";
import "mutationobserver-shim";
import Login from "./Login";
import { BrowserRouter } from "react-router-dom";
import moxios from "moxios";

let mockResponse = {
  status: 200,
  response: [
    {
      id: 1,
      email: "subro@gmail.com",
      password: "123456",
      firstname: "subro",
      lastname: "pattanaik",
      location: "pune",
      mobileno: "2312312412",
      select: false,
    },
    {
      id: 2,
      email: "bro@gmail.com",
      password: "123456",
      firstname: "bro",
      lastname: "pattanaik",
      location: "pune",
      mobileno: "2312312412",
      select: false,
    },
    {
      id: 3,
      email: "hr@gmail.com",
      password: "asdasd",
      firstname: "hrishi",
      lastname: "punde",
      location: "pune",
      mobileno: "2312312412",
      select: false,
    },
  ],
};

describe("Login Page Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter>
        <Login.WrappedComponent />
      </BrowserRouter>
    );
  });
  it("should test login class in container", () => {
    expect(wrapper.find(".login").exists()).toEqual(true);
  });

  it("should test empty class component", () => {
    expect(wrapper.find("other-class").exists()).toEqual(false);
  });

  it("should test Form component", () => {
    expect(wrapper.find("Form")).toHaveLength(1);
  });
});

describe("App Component Mount Test ", () => {
  let mountWrapper;

  beforeAll(() => {
    moxios.install();
  });

  afterAll(() => {
    moxios.uninstall();
  });

  it("renders correctly", (done) => {
    moxios.stubRequest("http://localhost:4000/users", mockResponse);
    mountWrapper = mount(
      <BrowserRouter>
        <Login.WrappedComponent />
      </BrowserRouter>
    );
    moxios.wait(() => {
      mountWrapper.update();
      expect(mountWrapper).toMatchSnapshot();
      done();
      mountWrapper.unmount();
    });
  });
});
