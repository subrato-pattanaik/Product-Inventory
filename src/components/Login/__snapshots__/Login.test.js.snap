// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Mount Test  renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Login>
      <Container
        className="p-0"
        fluid={true}
      >
        <div
          className="p-0 container-fluid"
        >
          <Row
            noGutters={true}
          >
            <div
              className="row no-gutters"
            >
              <Col
                lg={6}
                md={6}
                style={
                  Object {
                    "backgroundImage": "url(image1.jpg)",
                    "backgroundPosition": "center center",
                    "backgroundSize": "cover",
                    "minHeight": "89vh",
                  }
                }
                xs={12}
              >
                <div
                  className="col-lg-6 col-md-6 col-12"
                  style={
                    Object {
                      "backgroundImage": "url(image1.jpg)",
                      "backgroundPosition": "center center",
                      "backgroundSize": "cover",
                      "minHeight": "89vh",
                    }
                  }
                />
              </Col>
              <Col
                lg={6}
                md={6}
                xs={12}
              >
                <div
                  className="col-lg-6 col-md-6 col-12"
                >
                  <Container
                    className="d-flex login bg-light align-items-center"
                    fluid={false}
                  >
                    <div
                      className="d-flex login bg-light align-items-center container"
                    >
                      <Row
                        className="mx-auto justify-content-center"
                        noGutters={false}
                      >
                        <div
                          className="mx-auto justify-content-center row"
                        >
                          <Col
                            md={10}
                            xs={10}
                          >
                            <div
                              className="col-md-10 col-10"
                            >
                              <h3
                                className="display-4 text-warning text-center font-weight-bold"
                              >
                                PI Login
                              </h3>
                              <p
                                className="text-muted text-center mb-4"
                              >
                                Lets do shopping from here with best deals
                              </p>
                              <Form
                                autoComplete="off"
                                inline={false}
                                onSubmit={[Function]}
                              >
                                <form
                                  autoComplete="off"
                                  className=""
                                  onSubmit={[Function]}
                                >
                                  <FormGroup
                                    className="mb-4"
                                    controlId="formBasicEmail"
                                  >
                                    <div
                                      className="mb-4 form-group"
                                    >
                                      <FormLabel
                                        className="small lead"
                                        column={false}
                                        srOnly={false}
                                      >
                                        <label
                                          className="small lead form-label"
                                          htmlFor="formBasicEmail"
                                        >
                                          Email address
                                        </label>
                                      </FormLabel>
                                      <FormControl
                                        className="rounded-pill"
                                        name="email"
                                        placeholder="Enter email"
                                        type="email"
                                      >
                                        <input
                                          className="rounded-pill form-control"
                                          id="formBasicEmail"
                                          name="email"
                                          placeholder="Enter email"
                                          type="email"
                                        />
                                      </FormControl>
                                    </div>
                                  </FormGroup>
                                  <FormGroup
                                    className="mb-4"
                                    controlId="formBasicPassword"
                                  >
                                    <div
                                      className="mb-4 form-group"
                                    >
                                      <FormLabel
                                        className="small lead"
                                        column={false}
                                        srOnly={false}
                                      >
                                        <label
                                          className="small lead form-label"
                                          htmlFor="formBasicPassword"
                                        >
                                          Password
                                        </label>
                                      </FormLabel>
                                      <FormControl
                                        className="rounded-pill"
                                        name="password"
                                        placeholder="Password"
                                        type="password"
                                      >
                                        <input
                                          className="rounded-pill form-control"
                                          id="formBasicPassword"
                                          name="password"
                                          placeholder="Password"
                                          type="password"
                                        />
                                      </FormControl>
                                    </div>
                                  </FormGroup>
                                  <Button
                                    active={false}
                                    className="btn-block text-white border-0 rounded-pill mb-4"
                                    disabled={false}
                                    type="submit"
                                    variant="warning"
                                  >
                                    <button
                                      className="btn-block text-white border-0 rounded-pill mb-4 btn btn-warning"
                                      disabled={false}
                                      type="submit"
                                    >
                                      Login here
                                    </button>
                                  </Button>
                                </form>
                              </Form>
                            </div>
                          </Col>
                        </div>
                      </Row>
                    </div>
                  </Container>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Container>
    </Login>
  </Router>
</BrowserRouter>
`;
